app/jobs/search_keyword_job.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class SearchKeywordJob < ApplicationJob
+  queue_as :scrapers
+
+  def perform(search_result_id)
+    search_result = SearchResult.find(search_result_id)
+    html_code = Search::SearchService.new(
+      keyword: search_result.keyword,
+      search_engine: search_result.search_engine
+    ).search
+
+    raise IcRailsHueyToby::Errors::ClientServiceError unless html_code
```

Comment:
Do not raise exceptions for control flows. Exceptions are not "exceptional" cases 😉 they do not replace if/else blocks. Check this article: https://mensfeld.pl/2015/03/exceptions-should-not-be-expected-stop-using-them-for-control-flow-or-any-other-logic-handling-in-ruby/ and the linked articles at the bottom. 

```suggestion
    return mark_as_failed(search_result) unless html_code 
```

We could have a method `mark_as_failed` as it will be used also in the rescue block 💡